dodo.py

```diff
@@ -0,0 +1,484 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+"""
+Info: doit file to automate development tasks
+
+Command to list all the available tasks in the command line:
+doit list
+
+Get information on a single task:
+doit info <task-name>
+
+Targeted run of an individual task:
+Examples:
+    $ doit docs_build
+"""
+
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Generator, Optional
+
+import plumbum
+from doit import task_params
+from doit.tools import CmdAction
+
+from tagging.git_helper import GitHelper
+
+docker = plumbum.local["docker"]
+
+# global doit config
+DOIT_CONFIG = {"verbosity": 2, "default_tasks": ["what"]}
+
+
+def task_what() -> dict[str, Any]:
+    """
+    Prints the help message for this file 📖
+    """
+    return dict(
+        actions=[
+            Utils.do(
+                "echo",
+                "Use `doit list` to see all available tasks or `doit help` for available commands",
+            )
+        ],
+    )
+
+
+# -----------------------------------------------------------------------------
+# Setup tasks
+# -----------------------------------------------------------------------------
+def task_setup_dev() -> dict[str, Any]:
+    """
+    Install libraries required for development 🔧
+    """
+    actions = [
+        Utils.do(
+            *Utils.PIP,
+            "--requirement",
+            Paths.REQS_DEV,
+        )
+    ]
+    if Utils.IS_CI:
+        actions.insert(0, Utils.do(*Utils.PIP, "--upgrade", "pip"))
+
+    return dict(actions=actions)
+
+
+def task_setup_docs() -> dict[str, Any]:
+    """
+    Install libraries required for building docs 🔧
+    """
+
+    actions = [
+        Utils.do(
+            *Utils.PIP,
+            "--requirement",
+            Paths.REQS_DOCS,
+        ),
+    ]
+    if Utils.IS_CI:
+        actions.insert(
+            0,
+            Utils.do(*Utils.PIP, "--upgrade", "pip"),
+        )
+
+    return dict(actions=actions)
```

Comment:
I don't think this is needed.
I think it's easier to tell people that they need to install `requirements.txt` than to tell them that they need to install doit first, then they have to run `doit setup_docs` to install other requirements.
It's what many python-based projects require.
The same goes for `task_setup_dev`